vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc

```diff
@@ -281,14 +281,14 @@ void MediaTwitter::SavePublisherInfo(
     return;
   }
 
-  ledger::VisitData visit_data;
-  visit_data.provider = TWITTER_MEDIA_TYPE;
-  visit_data.url = url;
-  visit_data.favicon_url = favicon_url;
-  visit_data.name = publisher_name;
+  ledger::VisitDataPtr visit_data = ledger::VisitData::New();
```

Comment:
Why do we need to do a pointer here? Why not same as we do it here https://github.com/brave/brave-core/pull/2581/files#diff-6904081a34e36c0d05014066ffff4775R324?

Reply:
it eventually needs to be converted to VisitDataPtr so why wouldn't we just create it that way?